pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +999,38 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// check whether zone endpoint list need to be updated or not
+func UpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) bool {
+	updateZoneEndpointList := false
+	isZonePresent := false
+	zoneEndpointsList := []string{}
+	checkZoneEndpoints := make(map[string]bool)
+	if zoneName != "" {
+		if len(zones) > 0 {
+			for _, z := range zones {
+				if z.Name == zoneName {
+					for _, endpoint := range z.Endpoints {
+						zoneEndpointsList = append(zoneEndpointsList, endpoint)
+						checkZoneEndpoints[endpoint] = true
+					}
+					isZonePresent = true
+					break
+				}
+			}
+		}
+		if isZonePresent {
+			if len(desiredEndpointList) != len(zoneEndpointsList) {
+				updateZoneEndpointList = true
+			} else {
+				for _, endpoint := range desiredEndpointList {
+					if !checkZoneEndpoints[endpoint] {
+						updateZoneEndpointList = true
+						break
+					}
+				}
+			}
+		}
+	}
+	return updateZoneEndpointList
+}
```

Comment:
After making a `findZoneEndpoints()` function and a function to determine if lists are the same, this function becomes much easier to write and read:

```go
func ShouldUpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) (bool, err) {
    if zoneName == "" {
        // error
    }
    zoneExists, endpoints := findZoneEndpoints(zoneName, zones)
    if !zoneExists {
        return false
    }
    return !listsAreEqual(desiredEndpointList, endpoints)
}
```